### ISSUE TITLE
Expo permissions module was removed in SDK 42

### ISSUE THREAD

**Issue by JB1905** [byCedric/use-expo#510]
### Description of the bug
Expo permissions module was removed in SDK 42. Permissions module should be splitted into separate hooks for given Expo APIs (camera, notifications, etc.)

**Comment by flexbox** [comment]
@JB1905 Do you think it will land somehow one day?

I have the feeling that @byCedric doesn't have time to maintain this repo —that's perfectly fine tho!

I can't migrate to the SDK42 because I am using this nice permission hook 😭

**Comment by byCedric** [comment]
I'll take a stab at a new permissions hook. One that accepts permission requesters and getters, since we deprecated the `expo-permissions` library.

It's not that I don't have time for this. We are trying to move over the most useful hooks into expo itself 😁

**Comment by flexbox** [comment]
>the most useful hooks into expo itself 😁
That's what I thought, this repo is an "experimental" place before rolling out to the main repo and gets tons of news issues 😂

**Comment by byCedric** [comment]
Exactly! But, that still means I have to fix the permissions hook for sdk 42 though 🙈 So I'll do that asap.

Instead of permission constants, it will accept the permission requesters itself. E.g. `Camera.requestMicrophonePermission`

**Comment by byCedric** [comment]
So for single permission requests, I can [imagine something working like this](https://snack.expo.dev/@bycedric/use-expo-permission-single-hook) (see the `use-permission.ts` file). To request multiple permissions, I'd have to play around with some generics etc. But this should at least give you a similar API to use in your project @flexbox.

**Comment by flexbox** [comment]
Thanks for the feedback @byCedric
I successfully migrated `Camera` and `Location` permissions with your hook!

**Comment by byCedric** [comment]
FYI, this hook is going to be part of Expo itself. If this is starting from SDK 43, and not 42, I will add a copy of that hook in this library.

**Comment by byCedric** [comment]
@flexbox @JB1905 I [created a PR for this internally](https://github.com/expo/expo/pull/13782), if you have any feedback feel free to comment on it 😄

**Comment by flexbox** [comment]
@byCedric Thanks I will have a look 🚀
